core/lib/page-functions.js

```diff
@@ -498,6 +498,12 @@ function getNodeDetails(element) {
   const selector = getNodeSelector(element);
 
   // Create an id that will be unique across all execution contexts.
+  //
+  // Made up of 3 components:
+  //   - number unique to specific execution context
```

Comment:
It would be great to document somewhere that we don't need this because we have multiple isolated contexts at the end of things, we just have the one. We only really need this number because some elements may have been saved by other gatherers in other execution contexts, and we need to be able to know they don't match any element saved by FPS because those contexts are long gone by the time FPS rolls around.

Functionally it's the same thing, but we currently make it sound more complicated than it really is (as if we're juggling multiple _simultaneous_ isolated execution contexts).

Reply:
But we do have multiple execution contexts - the page context and the optional isolated context.

Reply:
Only the one _isolated_ context. Equivalently, we only have one context ID at a time, including at the end of the run.